app/models/route/RouteTable.scala

```diff
@@ -43,6 +43,10 @@ object RouteTable{
     val route = routes.filter(_.routeId === routeId).list
     route.headOption
   }
+  def getRegionByRouteId(routeId: Option[Int]): Option[Int] = db.withSession { implicit session =>
+    val regionId = routes.filter(_.routeId === routeId).list.map(_.regionId).headOption
```

Comment:
You can select a single column using slick instead of doing select * then filtering.
routes.filter(_.routeId === routeId).map(_.regionId).list.headOption